rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -334,10 +339,66 @@ where
             mailbox_address: self.address(),
             mailbox_domain: self.domain.id(),
             root: root.into(),
-            index,
+            index
         })
     }
 
+    #[instrument(level = "debug", err, ret, skip(self))]
+    #[allow(clippy::needless_range_loop)]
+    async fn tree(&self, lag: Option<NonZeroU64>) -> ChainResult<IncrementalMerkle> {
+        let lag = if let Some(lag) = lag { lag.get() } else { 0 }.into();
```

Comment:
I think this might be cleaner than what I suggested earlier.
```suggestion
        let lag = lag.map(|v| v.get()).unwrap_or(0).into();
```